core/signature_receiver.go

```diff
@@ -0,0 +1,329 @@
+package core
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+	"slices"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+// signatureReceiver is a struct for receiving SigningMessages for a single batch. It should never be instantiated
+// manually: it exists only as a helper struct for the ReceiveSignatures method.
+type signatureReceiver struct {
+	logger               logging.Logger
+	indexedOperatorState *IndexedOperatorState
+
+	signerMap                 map[OperatorID]bool
+	aggregateSignatures       map[QuorumID]*Signature
+	aggregateSignersG2PubKeys map[QuorumID]*G2Point
+
+	stakeSigned map[QuorumID]*big.Int
+
+	batchHeaderHash    [32]byte
+	signingMessageChan chan SigningMessage
+	quorumIDs          []QuorumID
+}
+
+// ReceiveSignatures receives SigningMessages over the signingMessageChan, and yields QuorumAttestations produced
+// from these SigningMessages.
+//
+// The yielded QuorumAttestations contain aggregate signing data from all SigningMessages received thus far. Each
+// QuorumAttestation will have incorporated more SigningMessages than the previously yielded QuorumAttestation.
+//
+// The caller should not manually close the returned QuorumAttestation chan. It will be closed automatically, when:
```

Comment:
It's ok to leave this comment. Just pointing out that there is no requirement to close a golang channel when you are done using it. In general, channels are only closed when doing so sends some sort of signal you want to transmit (e.g. "we're done, go away now"). So even without this comment, in general I wouldn't think that closing this channel from the outer context would be something people would be likely to accidentally do.

Reply:
I simplified the comment, to avoid saying something redundant:

```
This channel will be closed when one of the following conditions is met:
```

Reply:
Right also afaiu it's always (?) the sender's job to close the channel. It's actually very dangerous for a receiver to close a channel because a write on a closed channel panics, so unless the receiver has communicated with the sender through some other means and is damn sure that the sender won't close it (can't think of a situation where this would be the case, since if you've communicated with the sender, then might as well ask him to close the channel for you), then better to leave sender close channel.